### ISSUE TITLE
I have my own data set, but I still have some problems with parameter adjustment.

### ISSUE THREAD

**Issue by hunkyu** [KOKIAOKI/3d_bbs#45]
I have my own data set, but I still have some problems with parameter adjustment.The sample data has a better effect. I simulated a set of self-built data here, but the effect is not good and the posture is not quite right. this data is
https://drive.google.com/file/d/1nZwvnqPMtU-kPsRrpo3b4af8yuYMURKl/view?usp=drive_link

**Comment by KOKIAOKI** [comment]
Thank you for your report. I'll try to adjust parameters. Could you accept access right of 180447001@ccalumni.meijo-u.ac.jp?

**Comment by hunkyu** [comment]
ok. this is my email . 513459395@qq.com

**Comment by KOKIAOKI** [comment]
I mean I cannot access the google drive that you past above, so could you accept my request? I sent the request just now.

**Comment by hunkyu** [comment]
The access rights of Google Cloud Disk have been turned on. Can you try again?

**Comment by KOKIAOKI** [comment]
I can download the dataset.

**Comment by hunkyu** [comment]
Can the given dataset be adapted?

**Comment by KOKIAOKI** [comment]
Yes. I've tried to align both data.
One of the data can be localized with default parameter, but another one including indoor environment failed to match.
This is because a small range of point cloud gets buried in a voxel large enough to contain it.

I will try to adjust again in the next free time, but unfortunately I don't have a critical solution for the mismatch to the continuous voxels.
As a temporary fix, I recommend cropping some of the forests and building contents.

**Comment by hunkyu** [comment]
Please give me a master version and the corresponding configuration, because I failed here.

**Comment by hunkyu** [comment]
Please provide us with your execution efficiency report.

**Comment by KOKIAOKI** [comment]
Here is a [config](https://github.com/KOKIAOKI/3d_bbs/blob/main/test/config/test.yaml)
Please show me the output in your command at first because I cannot grasp process there.

**Comment by hunkyu** [comment]
################[Necessary]################
## Folder paths
target_clouds: "/home/hunkyu/sparse4d/letian_data/target"
source_clouds: "/home/hunkyu/sparse4d/letian_data/source"
output_folder: "/home/hunkyu/sparse4d/letian_data/output"

## 3D-BBS parameters
min_level_res: 1.0 # [m], e.g. outdoor: 1.0, indoor: 0.5
max_level: 6

## Angular search range [rad]
# 6.28 input is converted to 2*M_PI
min_rpy: [-0.02,-0.02,0.0] # [roll, pitch, yaw]
max_rpy: [0.02,0.02,6.28] # [roll, pitch, yaw]

## Score threshold
# score_threshold = floor(src_points.size() * score_threshold_percentage)
score_threshold_percentage: 0.9

################[Optional]################
## Downsample target clouds [m]
tar_leaf_size: 0.1 # off: 0.0

## Downsample source clouds [m]
src_leaf_size: 2.0 # off: 0.0

## Crop source clouds [m]
min_scan_range: 0.0 # [m]
max_scan_range: 100 # [m]
# off: set 0.0 to both min_scan_range and max_scan_range

## Timeout [msec]
# Enter the number as int
timeout_msec: 0 # off: 0
 
## Final align (evaluate the error)
use_gicp: false 

./cpu_test /home/hunkyu/3d_bbs/test/config/test.yaml
[YAML] Loading paths...
[YAML] Loading 3D-BBS parameters...
[YAML] Loading angular search range...
[YAML] Loading score threshold percentage...
[YAML] Loading downsample parameters...
[Setting] Loading target pcds...
[Setting] Loading source pcds...
[Setting] Create output folder with date...
[Voxel map] Creating hierarchical voxel map...
[Voxel map] Execution time: 1002.28[msec] 
-------------------------------
[Localize] pcd file name: 1688989026900
[Localize] Execution time: 3912.51[msec] 
[Localize] Score: 335
-------------------------------
[Localize] pcd file name: 1688991499197
[Localize] Execution time: 10738.1[msec] 
[Localize] Score: 1291
[Localize] Average time: 7325.29[msec] per frame
![2024-10-27 15-21-07 的屏幕截图](https://github.com/user-attachments/assets/3356dd6f-9a8d-441c-8dfb-8b20a1207398)

**Comment by KOKIAOKI** [comment]
I get a same command output, but it doesn't result in such an output point cloud.
This point cloud aligned correctly on my side...

**Comment by hunkyu** [comment]
It has been solved. It is the problem I displayed. But there is still a problem [Voxel map] Creating hierarchical voxel map...
[Voxel map] Execution time: 38021.3[msec]
It takes a very long time. How can I optimize it?

**Comment by KOKIAOKI** [comment]
Have you utilized [this](https://github.com/KOKIAOKI/3d_bbs/blob/main/test/test_code.md#5-optional-load-voxelmap-coordinates-directly)?
Voxelmap coords can be previously saved and directly input when BBS run.

**Comment by hunkyu** [comment]
![webwxgetmsgimg](https://github.com/user-attachments/assets/5cacca51-657d-4d46-a22b-c25383e11e71)
![1988687463](https://github.com/user-attachments/assets/6c833790-7423-4230-af36-8528c2736d0a)
Yes, now it is the result of saver, but the time still does not come down, is it because the map is too big? Or is there too many current frames?

**Comment by hunkyu** [comment]
I recently tried to use data from a new city road and found that this problem exists on a large scale and takes a very long time. Can I contact you by email?

**Comment by KOKIAOKI** [comment]
Has the loading time of the voxel map decreased?
The processing time for localization is a separate issue from saving the voxel map.

It's possible, but I can only analyze the data when I have time.

**Comment by hunkyu** [comment]
Yes, the map has been significantly improved, but the positioning time is still relatively long. I will close this issue first and have contacted
